scanomatic/generics/abstract_model_factory.py

```diff
@@ -1181,26 +1174,26 @@ def serialize_structure(obj, structure):
             )
 
     @staticmethod
-    def serialize(obj, dtype):
+    def serialize(obj, dtype) -> Optional[str]:
         if obj is None:
             return None
 
         elif isinstance(dtype, type) and issubclass(dtype, Enum):
-            return obj.name
+            return str(obj.name)
 
         elif dtype is _SectionsLink:
-            return pickle.dumps(obj)
+            return pickle.dumps(obj).decode('iso-8859-1')
 
         elif dtype in (int, float, str, bool):
             return str(obj)
 
         elif isinstance(dtype, types.FunctionType):
-            return pickle.dumps(dtype(serialize=obj))
+            return pickle.dumps(dtype(serialize=obj)).decode('iso-8859-1')
 
         else:
             if not isinstance(obj, dtype):
                 obj = dtype(obj)
-            return pickle.dumps(obj)
+            return pickle.dumps(obj).decode('iso-8859-1')
```

Comment:
I don't think these are reasonable, at least not without proper encode.

Reply:
I have some weak memory of having to resort to latin 1, so it might actually not be wrong... were there tests with old pickled things that were actually loaded when decoded like this?

Reply:
I think so, but I think we should take a short session to discuss this in person, so we can be reasonably sure we are on the same page.